homeassistant/components/tuya/climate.py

```diff
@@ -0,0 +1,463 @@
+        temp_set_value_range = json.loads(
+            self.tuya_device.status_range.get(dp_temp_set).values
+        )
+        return temp_set_value_range.get("scale")
+
+    def get_temp_current_scale(self) -> int:
+        """Get temperature current scale."""
+        dp_temp_current = (
+            DPCODE_TEMP_CURRENT if self.is_celsius() else DPCODE_TEMP_CURRENT_F
+        )
+        temp_current_value_range = json.loads(
+            self.tuya_device.status_range.get(dp_temp_current).values
+        )
+        return temp_current_value_range.get("scale")
+
+    # Functions
+
+    def set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        commands = []
+        if hvac_mode == HVAC_MODE_OFF:
+            commands.append({"code": DPCODE_SWITCH, "value": False})
+        else:
+            commands.append({"code": DPCODE_SWITCH, "value": True})
+
+        for tuya_mode, ha_mode in TUYA_HVAC_TO_HA.items():
+            if ha_mode == hvac_mode:
+                commands.append({"code": DPCODE_MODE, "value": tuya_mode})
+                break
+
+        self._send_command(commands)
+
+    def set_fan_mode(self, fan_mode: str) -> None:
+        """Set new target fan mode."""
+        self._send_command([{"code": DPCODE_FAN_SPEED_ENUM, "value": fan_mode}])
+
+    def set_humidity(self, humidity: float) -> None:
+        """Set new target humidity."""
+        self._send_command([{"code": DPCODE_HUMIDITY_SET, "value": int(humidity)}])
+
+    def set_swing_mode(self, swing_mode: str) -> None:
+        """Set new target swing operation."""
+        if swing_mode == SWING_BOTH:
+            commands = [
+                {"code": DPCODE_SWITCH_VERTICAL, "value": True},
+                {"code": DPCODE_SWITCH_HORIZONTAL, "value": True},
+            ]
+        elif swing_mode == SWING_HORIZONTAL:
+            commands = [
+                {"code": DPCODE_SWITCH_VERTICAL, "value": False},
+                {"code": DPCODE_SWITCH_HORIZONTAL, "value": True},
+            ]
+        elif swing_mode == SWING_VERTICAL:
+            commands = [
+                {"code": DPCODE_SWITCH_VERTICAL, "value": True},
+                {"code": DPCODE_SWITCH_HORIZONTAL, "value": False},
+            ]
+        else:
+            commands = [
+                {"code": DPCODE_SWITCH_VERTICAL, "value": False},
+                {"code": DPCODE_SWITCH_HORIZONTAL, "value": False},
+            ]
+
+        self._send_command(commands)
+
+    def set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        _LOGGER.debug("climate temp-> %s", kwargs)
+        code = DPCODE_TEMP_SET if self.is_celsius() else DPCODE_TEMP_SET_F
+        self._send_command(
+            [
+                {
+                    "code": code,
+                    "value": int(
+                        kwargs["temperature"] * (10 ** self.get_temp_set_scale())
```

Comment:
Is it not possible to set the high or low temperature?

Reply:
get_temp_set_scale() determined the temperature range